lessons/en/chapter_10.yaml

```
@@ -1,10 +1,149 @@
+
+    > THe command with `>>` will add text at the end of the file, while the operator `>` will override the file.
+  code: >-
+    https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&code=use+std%3A%3Aio%3B%0A%0Afn+main%28%29+%7B%0A++++let+mut+input+%3D+String%3A%3Anew%28%29%3B%0A%0A++++%2F%2F+the+read+will+be+stopped+by+%60%5Cn%60+character%0A++++if+let+Ok%28_%29+%3D+io%3A%3Astdin%28%29.read_line%28%26mut+input%29+%7B%0A++++++++println%21%28%22Input+text+%3A+%7B%7D%22%2C+input%29%3B%0A++++%7D%0A%7D%0A
+
+- title: File Descriptors
+  content_markdown: >
+    Now that we know what are the basic `I/O` operations, we dive even deeper.
+
+    You've already seen before: `stdin` (standard input), `stdout` (standard output) and `stderr` (stand error).
+    For each of them it is associated a positive integer number, a unique identifier
+    for an `I/O` channel (example: file), known as a `file descriptor (fd)`.
 
-    * [The Official Rust Programming
-    Book](https://doc.rust-lang.org/stable/book/)
+    Therefore:
+    - `stdin`: 0
+    - `stdout`: 1
+    - `stderr`: 2
+    - files
+
+    [![file descriptors](https://cs-pub-ro.github.io/operating-systems/assets/images/file-descriptors-d19424f0a417ecd1c032f98a1969ad75.svg)
+
+
+    Working with files plays an important role in the `I/O` operations.
+
+    You've already learned to open a text file and read its content.
+
+    But how about writing data in it, as an `I/O` channel.
+  code: >-
+    https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&code=use+std%3A%3Afs%3A%3A%7Bself%2C+File%7D%3B++++++%2F%2F+file+system%0Ause+std%3A%3Aio%3A%3A%7Bself%2C+Write%7D%3B+++++%2F%2F+input+output%0A%0Afn+main%28%29+-%3E+io%3A%3AResult%3C%28%29%3E+%7B%0A++++let+file_name+%3D+%22output.txt%22%3B%0A%0A++++%2F%2F+creates+the+file+if+it+doesn%27t+already+exists%0A++++let+mut+file+%3D+File%3A%3Acreate%28file_name%29%3F%3B%0A%0A++++let+text_to_write+%3D+%22Hello%2C+World%21%5Cn%5C%0A++++++++++++++++++++++++++++This+is+a+line+of+text.%5Cn%22%3B%0A++++file.write_all%28text_to_write.as_bytes%28%29%29%3F%3B%0A%0A++++let+absolute_path+%3D+fs%3A%3Acanonicalize%28file_name%29%3F%3B%0A++++println%21%28%22Text+has+been+written+to%3A+%7B%3A%3F%7D%22%2C+absolute_path%29%3B%0A%0A++++return+Ok%28%28%29%29%3B%0A%7D%0A
+
+- title: System Arguments
+  content_markdown: >
+    A Rust program is able to receive `input` from the in-line arguments.
+
+    In order to do so, open a [terminal](https://www.youtube.com/watch?v=lZ7Kix9bjPI), compile it and pass the arguments to the executable in the command promt.
+
+    These arguments might be relevant files, flags and so on.
+    The developer must document their purpose.
+
+    If you are using LINUX or macOS, bear in mind these commands:
+    ```bash
+    $ touch main.rs
+    $ rustc main.rs
+    $ ./main.rs 2 3 4 5 
+    ```
+
+    Otherwise, for Windows environment, on a PowerShell and paste them
```

Did you mean "open" instead of "on a"?
```suggestion
    Otherwise, for Windows environment, open PowerShell and paste them
```